### PR TITLE
OP-1309 | Create endpoints for CRUD on GroupPermissions

### PR DESCRIPTION
**Pull request by SilverD3**
See OP-1309

### Updates
- update endpoint `POST /users/groups` to accept a DTO containing the list of `PermissionDTO` to assign to user group being created.
- update endpoint `PUT/users/groups` to accept a DTO containing the updated list of `PermissionDTO` to assign to user group being updated.

### New
- e2e tests for `UserController`
- `POST /users/groups/{group_code}/permissions/{id}` to assign a permission to a user group
- `DELETE /users/groups/{group_code}/permissions/{id}` to revoke a permission from a user group

### How to test
Use the PR [1391](https://github.com/informatici/openhospital-core/pull/1391) on the CORE.
1. Run the app
2. Check if tests are passing
3. Use swagger to manually test the two updated endpoints

### CODE REVIEW

**Comment by SilverD3** [comment]
Test are failing because [1391 ](https://github.com/informatici/openhospital-core/pull/1391)is not merged.

File: src/test/java/org/isf/menu/data/UserGroupHelper.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+import org.isf.menu.TestUserGroup;
+import org.isf.menu.dto.UserGroupDTO;
+import org.isf.menu.model.UserGroup;
+import org.isf.utils.exception.OHException;
+
+import java.util.List;
```
Comment (dbmalkovsky) [inline src/test/java/org/isf/menu/data/UserGroupHelper.java]: The ordering of the `imports` does not match project standards.

File: src/main/java/org/isf/menu/rest/UserController.java [hunk 2]
```diff
@@ -256,23 +274,50 @@ public ResponseEntity<Boolean> newUserGroup(@Valid @RequestBody UserGroupDTO aGr
...
 		if (isUpdated) {
-			return ResponseEntity.ok(isUpdated);
+			return ResponseEntity.ok(true);
 		} else {
 			throw new OHAPIException(new OHExceptionMessage("User group not updated."));
 		}
 	}
 
+	@GetMapping(value = "/users/groups/{group_code}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public ResponseEntity<UserGroupDTO> getUserGroup(@PathVariable("group_code") String code) throws OHServiceException {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/rest/UserController.java:311]: Please add javadoc for this new method; thanks

**Comment by mwithi** [comment]
> Test are failing because [1391](https://github.com/informatici/openhospital-core/pull/1391)is not merged.

I'm testing locally with [1391](https://github.com/informatici/openhospital-core/pull/1391) and this test is failing:

```
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   UserControllerTest$UserGroupPermissions.revokePermissionFromUserGroup:815 Status expected:<200> but was:<403>
[INFO] 
[ERROR] Tests run: 143, Failures: 1, Errors: 0, Skipped: 0
```

File: src/test/java/org/isf/menu/rest/UserControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,845 @@
...
+							post("/users/groups/{group_code}/permissions/{id}", userGroup.getCode(), permission.getId())
+									.contentType(MediaType.APPLICATION_JSON)
+					)
+					.andDo(log())
+					.andExpect(status().is4xxClientError())
+					.andExpect(content().string(containsString("not found")));
+		}
+
+		@Test
+		@WithMockUser(username = "admin", authorities = { "usergroups.create", "grouppermission.create" })
```
Comment (mwithi) [inline src/test/java/org/isf/menu/rest/UserControllerTest.java]: Should this be like?

```suggestion
		@WithMockUser(username = "admin", authorities = { "grouppermission.delete" })
```

Comment (mwithi) [reply src/test/java/org/isf/menu/rest/UserControllerTest.java]: Hi @SilverD3, what about this and the failing test above?

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**